### ISSUE TITLE
How to add page flip animation for a touch device?

### ISSUE THREAD

**Issue by ahayder** [futurepress/epub.js#510]
Guys, please let me know how can I implement a page flip animation into EpubJs. Is it possible? Should I use another plugin?
You know there are no plugins close to this. :(

**Comment by geek1011** [comment]
I finally figured it out! 

````javascript
EPUBJS.Hooks.register('beforeChapterDisplay').pageAnimation = function (callback, renderer) {
    var style = renderer.doc.createElement("style");
    style.innerHTML = "*{-webkit-transition: all {t} ease;-moz-transition: all {t} ease;-o-transition: all {t} ease;-ms-transition: all {t} ease;transition: all {t} ease;}";
    style.innerHTML = style.innerHTML.replace("{t}", "0.5s");
    renderer.doc.body.appendChild(style);
    if (callback) {
        callback();
    }
};
````

**Comment by geek1011** [comment]
Also, at the moment, to support desktop and laptop devices, epub.js needs to be patched by adding the following code somewhere before you start using epub.js, but after you load the library. The only difference this makes is that epub.js will use css transforms on desktop devices as well as mobile devices to move the pages.

````javascript
EPUBJS.Render.Iframe.prototype.setLeft = function(leftPos){
    this.docEl.style[this.transform] = 'translate('+ (-leftPos) + 'px, 0)';
}
````

**Comment by georgezouq** [comment]
@geek1011   Why there need some code this:
```
(10,function(){parent.ePubViewer.Book.nextPage()},function(){parent.ePubViewer.Book.prevPage()});
```
If I make it work, It will turn two pages at a time

**Comment by geek1011** [comment]
Its to detect swipe. It seems like you already have another code to check for the swipe event

**Comment by carlosrc** [comment]
@geek1011 I have problems when transitions are enabled. When I resize the page, it goes crazy and presents strange behaviour, changing the current page. 

Anyone have this problem too?
Thanks.

**Comment by geek1011** [comment]
Yes, this can be worked around by checking for a resize, then disabling the animation temporarily.

On Mar 30, 2017 7:02 AM, Carlos Rodríguez <notifications@github.com> wrote:

@geek1011<https://github.com/geek1011> I have problems when transitions are enabled. When I resize the page, it goes crazy and presents strange behaviour, changing the current page.

Anyone have this problem too?
Thanks.

**Comment by carlosrc** [comment]
Could you tell me how to check for a resize? I can't find the way.

I appreciate your help. Thanks!

**Comment by carlosrc** [comment]
@geek1011 could you tell me how to check for a resize? I can't find the way.

I appreciate your help. Thanks!
